jina/types/document/__init__.py

```diff
@@ -457,8 +458,40 @@ def get_attrs(self, *args) -> Dict[str, Any]:
         .. seealso::
             :meth:`update` for bulk set/update attributes
 
+        .. note::
+            Arguments will be extracted using `dunder_get`
+            .. highlight:: python
+            .. code-block:: python
+
+                d = Document({'id': '123', 'hello': 'world', 'tags': {'id': 'external_id', 'good': 'bye'}})
+
+                assert d.id == '123'  # true
+                assert d.tags['hello'] == 'world' # true
+                assert d.tags['good'] == 'bye' # true
+                assert d.tags['id'] == 'external_id' # true
+
+                res = d.get_attrs(*['id', 'tags__hello', 'tags__good', 'tags__id'])
+
+                assert res['id'] == '123' # true
+                assert res['tags__hello'] == 'world' # true
+                assert res['tags__good'] == 'bye' # true
+                assert res['tags__id'] == 'external_id' # true
         """
-        return {k: getattr(self, k) for k in args if hasattr(self, k)}
+
+        ret = {}
+        for k in args:
+            try:
```

Comment:
`hasattr()` can be used instead of try except

Reply:
Agree. `hasattr` is a better option here.

Reply:
I did it for potential performance when I was checking access in tags, but is true for just Document is better to do so